src/lib/y2storage/dialogs/guided_setup/helpers/candidate_disks.rb

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) [2019] SUSE LLjkkC#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+
+module Y2Storage
+  module Dialogs
+    class GuidedSetup
+      module Helpers
+        # Helper that provides a handful of methods to query information related to candidate disks,
+        # mainly about partitions.
+        class CandidateDisks
+          def initialize(settings, analyzer)
+            @settings = settings
+            @analyzer = analyzer
+          end
+
+          # Convenience method to expose the candidate disks
+          #
+          # @return [Array<Disks>]
+          def all
```

Comment:
NP: I am not sure it is clear what the method returns when calling to `candidate_disks_helper.all`. Most likely, I would have to check the method definition to be sure what it is returning. IMHO, `candidate_disks_helper.candidate_disks` is more explicit. But I am fine with your proposal, you can keep it if you want ;)

Reply:
:thinking: Probably you're right. We don't know at first glance if `.all` is `all candidate disks` or `all partitions` or `all whatever`.

However, `candidate_disks_helpers.candidate_disks` looks a bit redundant to me. And long, like the `candidate_disks_helper.single_candidate_disk?` one.

Anyway, I am going to follow your suggestion and I'll postpone the search of better method names for the next iteration over this piece of code.

Thanks!